src/Command/Service/CreateCoreQueryArchCommandService.php

```diff
@@ -0,0 +1,439 @@
+                true
+            );
+            $output->writeln(
+                '<info>Using Existing Response : ' . $this->responseDtoTemplate->namespace . '\\'
+                    . $this->responseDtoTemplate->name . '</info>'
+            );
+        }
+        $output->writeln('<comment>' . $this->getRelativeFilePath($filePath) . '</comment>');
+        $output->writeln("");
+    }
+
+    public function createPresenterInterfaceIfNotExist(
+        OutputInterface $output,
+        string $modelName,
+        string $useCaseType
+    ): void {
+        $useCaseName = $useCaseType . $modelName;
+        $className = $useCaseName . 'PresenterInterface';
+        $filePath = $this->srcPath . '/Core/' . $modelName . '/Application/UseCase/' . $useCaseName . '\\'
+            . $className . '.php';
+        $namespace = 'Core\\' . $modelName . '\\Application\\UseCase\\' . $useCaseName;
+        if (!file_exists($filePath)) {
+            $this->commandPresenterInterfaceTemplate = new PresenterInterfaceTemplate(
+                $filePath,
+                $namespace,
+                $className,
+                false
+            );
+            preg_match('/^(.+).' . $className . '\.php$/', $filePath, $matches);
+            $dirLocation = $matches[1];
+            //Create dir if not exists,
+            if (!is_dir($dirLocation)) {
+                mkdir($dirLocation, 0777, true);
+            }
+            file_put_contents(
+                $this->commandPresenterInterfaceTemplate->filePath,
+                $this->commandPresenterInterfaceTemplate->generateModelContent()
+            );
+            $output->writeln(
+                '<info>Creating Presenter Interface : ' . $this->commandPresenterInterfaceTemplate->namespace . '\\'
+                    . $this->commandPresenterInterfaceTemplate->name . '</info>'
+            );
+        } else {
+            $this->commandPresenterInterfaceTemplate = new PresenterInterfaceTemplate(
+                $filePath,
+                $namespace,
+                $className,
+                true
+            );
+            $output->writeln(
+                '<info>Using Existing Presenter Interface : ' . $this->commandPresenterInterfaceTemplate->namespace . '\\'
+                    . $this->commandPresenterInterfaceTemplate->name . '</info>'
+            );
+        }
+        $output->writeln('<comment>' . $this->getRelativeFilePath($filePath) . '</comment>');
+        $output->writeln("");
+    }
+
+    public function createPresenterIfNotExist(
+        OutputInterface $output,
+        string $modelName,
+        string $useCaseType
+    ): void {
+        $useCaseName = $useCaseType . $modelName;
+        $className = $useCaseName . 'Presenter';
+        $filePath = $this->srcPath . '/Core/' . $modelName . '/Infrastructure/Api/' . $useCaseName . '\\'
+            . $className . '.php';
+        $namespace = 'Core\\' . $modelName . '\\Infrastructure\\Api\\' . $useCaseName;
```

Comment:
By default, all the new **`API`** namespace are named API and not **`Api`**.
```suggestion
        $filePath = $this->srcPath . '/Core/' . $modelName . '/Infrastructure/API/' . $useCaseName . '\\'
            . $className . '.php';
        $namespace = 'Core\\' . $modelName . '\\Infrastructure\\API\\' . $useCaseName;
```